examples/proc_GDS_pulse_capture.py

```diff
@@ -0,0 +1,121 @@
+Assuming the data is the capture of the pulse sequence as seen on the GDS
+oscilloscope, here the data is converted to analytic power, frequency filtered
+and the absolute is taken prior to integrating to return the beta value (us
+sqrt(W)). 
+"""
+import pyspecdata as psd
+import matplotlib.pyplot as plt
+from numpy import sqrt, array
+
+show_all = False
+with psd.figlist_var() as fl:
+    for filename, nodename, beta90_range, beta180_range in [
+        (
+            "240729_test_p90_pulse_calib.h5",
+            "pulse_calib_2",
+            (None, 34e-6),
+            (34e-6, None),
+        ),
+    ]:
+        d = psd.nddata_hdf5(
+            filename + f"/{nodename}",
+            directory=psd.getDATADIR(exp_type="ODNP_NMR_comp/test_equipment"),
+        )
+        beta_1 = []
+        beta_2 = []
+        for j in range(len(d["p_90"])):
+            s = d["p_90", j].C
+            s.set_units("t", "s")
+            s *= 101.35  # attenutation ratio
+            s /= sqrt(2)  # Vrms
+            s /= sqrt(50)  # V/sqrt(R) = sqrt(P)
+            if show_all:
+                fl.basename = r"programmed $\beta$ = %s" % str(
+                    s.get_prop("set_beta")[j]
+                )
+                fig, ax_list = plt.subplots(1, 2, figsize=(10, 4))
+                fig.suptitle(fl.basename)
+                # {{{ Plot raw capture
+                ax_list[1].set_ylabel(r"$\sqrt{P_{pulse}}$")
+                ax_list[0].set_ylabel(r"$\sqrt{P_{pulse}}$")
+                ax_list[1].set_ylabel(None)
+                ax_list[1].set_title("180 Pulse")
+                fl.next("Pulse Calibration")
+                fl.plot(s["t":beta90_range], alpha=0.2, color="blue", ax=ax_list[0])
+                fl.plot(s["t":beta180_range], alpha=0.2, color="blue", ax=ax_list[1])
+                # }}}
+            # {{{ make analytic
+            s.ft("t", shift=True)
+            s = s["t":(0, None)]
+            s *= 2
+            s["t":0] *= 0.5
+            # }}}
+            if show_all:
+                s.ift("t")
+                # {{{ Plot analytic
+                fl.plot(abs(s["t":beta90_range]), color="orange", ax=ax_list[0])
+                fl.plot(abs(s["t":beta180_range]), color="orange", ax=ax_list[1])
+                s.ft("t")
+                # }}}
+            # {{{ frequency filter
+            s["t":(0, 11e6)] *= 0
+            s["t":(24e6, None)] *= 0
+            # }}}
+            s.ift("t")
+            if show_all:
+                # {{{ plot and integrate 90 pulse
+                fl.plot(abs(s["t":beta90_range]), ax=ax_list[0], color="red")
+                fl.plot(abs(s["t":beta180_range]), ax=ax_list[1], color="red")
+                # }}}
+            beta90_int = abs(s["t":beta90_range]).contiguous(
+                lambda x: x > 0.01 * s["t":beta90_range].max()
+            )[0]
+            beta180_int = abs(s["t":beta180_range]).contiguous(
+                lambda x: x > 0.01 * s["t":beta180_range].max()
```

Comment:
select integration range based on lambda function (anythin higher than 0.01 of the max in the selected region). Selected region needed to separate the 90 pulse from the 180 pulse

Reply:
same as above + see comment about expanding

Reply:
(although this what does "based on lambda function" mean? You're just using the `contiguous` function as it's intended to be used)